torch/csrc/jit/tensorexpr/types.cpp

```diff
@@ -9,6 +9,10 @@ namespace torch {
 namespace jit {
 namespace tensorexpr {
 
+static bool is_c10_type(const ScalarType& type) {
+  return type < ScalarType::Undefined;
```

Comment:
Minor: this is fine. But it feels a bit dangerous to rely on enum value ordering. I've seen in other systems when a list of enums is imported, it also uses the same macros to define a list of type traits on those enums. Otherwise, the enum value might keep changing, which makes serialization compatibility a bit complicated. It is up to you.

Reply:
Right, so the main reason I want to use ordering here is that we define the first N values of te::ScalarType in lockstep with c10::ScalarType (and we depend on that all over the place).  Maybe something less brittle would be better.  But I think I'd rather put the effort towards making te::ScalarType just go away in favor of c10::ScalarType, if possible. :)